dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java

```diff
@@ -334,15 +335,23 @@ public void destroy() {
 
     @Override
     public URL getUrl() {
+        URL getUrl;
         if (currentAvailableInvoker != null) {
-            return currentAvailableInvoker.getUrl();
+            getUrl = currentAvailableInvoker.getUrl();
         } else if (invoker != null) {
-            return invoker.getUrl();
+            getUrl = invoker.getUrl();
         } else if (serviceDiscoveryInvoker != null) {
-            return serviceDiscoveryInvoker.getUrl();
+            getUrl = serviceDiscoveryInvoker.getUrl();
+        } else {
+            getUrl = consumerUrl;
         }
 
-        return consumerUrl;
+        // set parameter weight if needed.
+        if (getUrl != null && getUrl.getUrlParam() != null && StringUtils.isEmpty(getUrl.getParameter(WEIGHT_KEY))
+                && url != null && url.getUrlParam() != null && !StringUtils.isEmpty(url.getParameter(WEIGHT_KEY))) {
+            getUrl = getUrl.addParameter(WEIGHT_KEY, url.getParameter(WEIGHT_KEY));
+        }
+        return getUrl;
```

Comment:
I think it would be more proper to move these to somewhere real registry invoker created.